src/Core/Services/StateService.cs

```diff
@@ -291,6 +291,20 @@ public async Task<bool> CanAccessPremiumAsync(string userId = null)
             return organizations?.Any(o => o.UsersGetPremium && o.Enabled) ?? false;
         }
 
+        public async Task SetPremiumAsync(bool value, string userId = null)
```

Comment:
I think we should use a different setter name here to make it clear that we're setting the `HasPremiumPersonally` value (and nothing to do with the org).  Maybe `SetPersonalPremiumAsync`?